src/check_pack.c

```diff
@@ -38,16 +38,33 @@
 #endif
 
 /* Maximum size for one message in the message stream. */
-#define CK_MAX_MSG_SIZE 8192
+static int ck_max_msg_size = 0;
 /* This is used to implement a sliding window on the receiving
  * side. When sending messages, we assure that no single message
- * is bigger than this (actually we check against CK_MAX_MSG_SIZE/2).
+ * is bigger than this.
  * The usual size for a message is less than 80 bytes.
  * All this is done instead of the previous approach to allocate (actually
  * continuously reallocate) one big chunk for the whole message stream.
  * Problems were seen in the wild with up to 4 GB reallocations.
  */
 
+void check_set_max_msg_size(int max_msg_size) {
+    ck_max_msg_size = max_msg_size;
+}
+
+static int get_max_msg_size(void) {
+    if (ck_max_msg_size <= 0) {
```

Comment:
To be consistent with the behavior of the other environment variables, the order of precedence should be:
1) environment variable
2) check_set_max_msg_size()
3) default value

See CK_DEFAULT_TIMEOUT as an example of this.